### ISSUE TITLE
TypeError: Cannot read property '2' of null

### ISSUE THREAD

**Issue by abartolo** [feathersjs/feathers-sequelize#71]
Hello,

So I'm giving FeathersJs a try and so far its supper easy to create quick rest apis. However I am stuck on an issue when creating a duplicate record where the table column is unique. I came across this issue when testing since other application might want to send a POST request to create a record but it "email" for example might already be used.

## What you are doing?
Creating a record with the same value in a column that is Unique:true. I successfully inserted new record via restful api using Postman.  However if I send the same request again( create a user with the same email which the column is unique) it crashes the server with error "TypeError: Cannot read property '2' of null".

Below is the model I am using

```js
// code here
const Sequelize = require('sequelize');

module.exports = function(sequelize) {
  const user = sequelize.define('users', {
    email: {
      type: Sequelize.STRING,
      allowNull: false,
      unique: true
    },
    password: {
      type: Sequelize.STRING,
      allowNull: false
    }
  }, {
    freezeTableName: true
  });
```

## What do you expect to happen?
I expected to get a response back with an error stating its a duplicate or at least a clean error.

## What is actually happening?
Crashed server since error wasn't caught.

## Error Image

![image](https://cloud.githubusercontent.com/assets/5394228/19537833/7c84e15c-961f-11e6-9f0e-ec5aaa9a72f3.png)

__Dialect:__ mysql
__Database version:__ 5.7.13
__Sequelize version:__ 3.24.4

**Comment by abartolo** [comment]
Update - Issue above was @ work. I tried to replicate @ home and got the following response. I will download dependencies @work again in case there was an issue I missed something.

__Dialect:__ mysql
__Database version:__ 5.6.25
__Sequelize version:__ 3.24.4

`
{
  "name": "BadRequest",
  "message": "Validation error",
  "code": 400,
  "className": "bad-request",
  "errors": [
    {
      "message": "test_description_UNIQUE must be unique",
      "type": "unique violation",
      "path": "test_description_UNIQUE",
      "value": "netace_edgeUpdate_creditCard_splitpayment_visa"
    }
  ]
}
`

**Comment by daffl** [comment]
Thank you for the detailed report. That last response looks more like what it should do. I'd say a dependency issue or a problem with the database.

**Comment by abartolo** [comment]
@daffl After reinstalling dependencies and seeing no issue - the same issue occurred :(. Sucks im really enjoying Feathers at home. super easy.

So after more analysis I found I looked at my node version found I'm using v4.2.1 @ work. This is a red flag since FeatherJS documentation is stating nodeV 5+ and if anything is below is required to use babel.

This is the NPM Error I get below the first image I attached.
![image](https://cloud.githubusercontent.com/assets/5394228/19562576/ddbd40aa-96aa-11e6-9d75-9ebda5fbe2f8.png)


Not sure if this is related to this issue but I will work on updating my node and see if that fixes the issue.

@daffl what are the chances this is a due to babel not executing properly.. I have a feeling because @ work security is very strict.. I can't even update node without following a process..

**Comment by daffl** [comment]
That does look like a Babel error. It should work with Node v4 but with strict security restrictions it makes sense something during installation went wrong.

I recommend using [nvm](https://github.com/creationix/nvm) to install the latest version and try that. You might want to re-initialize the application so that it doesn't enable Babel (this is only done for Node < 5).

**Comment by abartolo** [comment]
@daffl It seems to be mysql version issue. I originally opened issue at sequelize repository but was initially closed assuming it wasn't a sequelize issue but then recently reopened. Luckily the mysql version locally installed at work is version 5.6.15 didn't display this issue VS. the mysql 5.7.13 where I got this issue.

I replicated both table structures and unique indexes as seen below. Sent a POST to create a new record and immediately sent the POST again to recreate the crash but this time the server did not crash and instead got a JSON error as expected.

Expected Response with mysql version 5.6.15
`
{
	"test_description": "b2b_edgeFlow_creditCard_visa_splitpayment_3",
	"status": "failed",
	"build_date": "2016-08-01 15:20:28"
}
`

Both have the same Table and Indexes
![image](https://cloud.githubusercontent.com/assets/5394228/19568893/75e6e550-96c1-11e6-8554-d936820598a0.png)


Indexes
![image](https://cloud.githubusercontent.com/assets/5394228/19568909/89dd2632-96c1-11e6-952c-796802e24bb9.png)
